### PR TITLE
[Go] add menu sample

### PR DESCRIPTION
**Pull request by ianlancetaylor**
This is based closely on the TypeScript code.

The s05 tests don't work yet. They get an error

LookupGeneratorAction: no generator action named "google-vertexai"/"gemini-1.0-pro-vision"

Using the gemini-1.0-pro action (without "-vision") doesn't work, saying that it doesn't support the input type.

Description here... 

Checklist (if applicable):
- [ ] Tested (manually, unit tested, etc.)
- [ ] Changelog updated
- [ ] Docs updated

### CODE REVIEW

File: go/samples/menu/s01.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			InputSchema:  menuQuestionInputSchema,
+			OutputFormat: ai.OutputFormatText,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (jba) [inline go/samples/menu/s01.go:80]: return err

File: go/samples/menu/s02.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+func menu(ctx context.Context, input map[string]any) (map[string]any, error) {
+	f, err := os.Open("testdata/menu.json")
+	if err != nil {
+		return nil, err
+	}
+	decoder := json.NewDecoder(f)
+	var s []any
+	if err := decoder.Decode(&s); err != nil {
+		return nil, err
```
Comment (jba) [inline go/samples/menu/s02.go:44]: Consider dropping `decoder`:

    var s []any
     if err := json.NewDecoder(f).Decode(...)

**Comment by jba** [comment]
Thanks, lot of work.

Did you find anything awkward that we could make smoother?

**Review by jba — CHANGES_REQUESTED** [review]

File: go/samples/menu/main.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+func main() {
+	projectID := os.Getenv("GCLOUD_PROJECT")
+	if projectID == "" {
+		fmt.Fprintln(os.Stderr, "menu example requires setting GCLOUD_PROJECT in the environment.")
+		os.Exit(1)
+	}
+
+	location := "us-central1"
+	if envLocation := os.Getenv("GCLOUD_LOCATION"); envLocation != "" {
```
[the remark this answers is not in the document]
Comment (ianlancetaylor) [reply go/samples/menu/main.go:80]: Used `env`.

File: go/samples/menu/s01.go (hunk 1 above)
Comment (ianlancetaylor) [reply go/samples/menu/s01.go:80]: Done.

**Review by jba — APPROVED** [review]

**Comment by ianlancetaylor** [comment]
> Did you find anything awkward that we could make smoother?

I think the most awkward parts I saw were the document handling.  Code like this in s05.go:

```Go
			presp, err := textMenuPrompt.Generate(ctx, preq, nil)
			if err != nil {
				return nil, err
			}
			ret := &answerOutput{
				Answer: presp.Candidates[0].Message.Content[0].Text,
			}
```

It's a long walk from `presp` to the actual data.

**Pull request merged**